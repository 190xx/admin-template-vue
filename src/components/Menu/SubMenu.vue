<template functional>
    <ASubMenu v-on="listeners" :key="data.name">
        <span slot="title"><AIcon type="mail" /><span>{{ data.attrs.item.title }}</span></span>
        <template v-for="i in data.attrs.item.children">
            <AMenuItem v-if="!i.children" :key="i.name">
                <AIcon v-if="i.icon" :type="i.icon" />
                <span>{{ i.title }}</span>
            </AMenuItem>
            <SubMenu v-else :item="i" :key="i.name" />
        </template>
    </ASubMenu>
</template>
