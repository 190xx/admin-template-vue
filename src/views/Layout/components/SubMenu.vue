<template functional>
  <ASubMenu :key="data.key" v-on="listeners">
    <template #title>
      <AIcon :type="props.item.icon" />
      <span>{{ props.item.title }}</span>
    </template>
    <template v-for="i of props.item.children">
      <AMenuItem v-if="!i.children" :key="i.key || i.name">
        <AIcon v-if="i.icon" :type="i.icon" />
        <span>{{ i.title }}</span>
      </AMenuItem>
      <SubMenu v-else :key="i.key || i.name" :item="i" />
    </template>
  </ASubMenu>
</template>
