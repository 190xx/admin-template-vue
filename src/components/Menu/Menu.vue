<template>
    <AMenu v-bind="$attrs">
        <template v-for="item in menuData">
            <AMenuItem v-if="!item.children" :key="item.name" @click="pushRouter">
                <AIcon v-if="item.icon" :type="item.icon" />
                <span>{{ item.title }}</span>
            </AMenuItem>
            <SubMenu v-else :item="item" :key="item.name" @click="pushRouter" />
        </template>
    </AMenu>
</template>

<script>
    import SubMenu from './SubMenu';

    export default {
        name: 'VMenu',
        components: { SubMenu },
        props: {
            menuData: {
                type: Array,
                required: true
            }
        },
        data () {
            return {
                list: []
            };
        },
        methods: {
            pushRouter(item) {
                this.$router.push({name : item.key})
            }
        }
    };
</script>

<style scoped>

</style>
