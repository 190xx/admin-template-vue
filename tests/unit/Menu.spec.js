import { mount } from '@vue/test-utils';
import Menu from '@c/Menu/Menu';
import SubMenu from '@c/Menu/SubMenu';

describe('Menu', () => {
    it('renders SubMenu passed', () => {
        const menuData = [
            {
                icon: 'pie-chart',
                name: 'Home',
                title: '测试1',
                children: [
                    {
                        icon: 'credit-card',
                        name: 'Home1',
                        title: '测试12'
                    }
                ]
            },
            {
                icon: 'setting',
                name: 'BlankPage2',
                title: '测试2'
            },
            {
                icon: 'pie-chart',
                name: 'Home',
                title: '测试1',
                children: [
                    {
                        icon: 'credit-card',
                        name: 'Home1',
                        title: '测试12'
                    }
                ]
            },
        ];
        const wrapper = mount(Menu, {
            stubs: {
                SubMenu: true,
                AMenu: true,
                AMenuItem: true,
                AIcon: true
            },
            propsData: { menuData }
        });
        expect(wrapper.findAll(SubMenu)).toHaveLength(2);
        expect(wrapper.html()).toMatchSnapshot();
    });
});
