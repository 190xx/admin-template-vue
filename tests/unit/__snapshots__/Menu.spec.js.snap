// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu.vue renders SubMenu passed 1`] = `
<amenu-stub>
  <submenu-stub item="[object Object]"></submenu-stub>
  <amenuitem-stub>
    <aicon-stub type="setting"></aicon-stub> <span>测试2</span>
  </amenuitem-stub>
  <submenu-stub item="[object Object]"></submenu-stub>
</amenu-stub>
`;
